src/main/java/com/gmail/goosius/siegewar/utils/CosmeticUtil.java

```diff
@@ -86,15 +86,16 @@ private static void changeBlocksToGlass(Player player, Location loc) {
 	}
 
 	public static void removeFakeBeacon(Player player, Location loc) {
-		player.sendBlockChange(loc.clone().subtract(0, 1, 0), Bukkit.createBlockData(loc.clone().subtract(0, 1, 0).getBlock().getType()));
-		player.sendBlockChange(loc.clone().subtract(0, 2, 0), Bukkit.createBlockData(loc.clone().subtract(0, 2, 0).getBlock().getType()));
-
-		int[][] ironBlockLocations = {{1, 1}, {1, 0}, {1, -1}, {0, 1}, {0, 0}, {0, -1}, {-1, 1}, {-1, 0}, {-1, -1}};
-		for (int i = 0; i < 9; i++) {
-			player.sendBlockChange(loc.clone().add(ironBlockLocations[i][0], -3, ironBlockLocations[i][1]), Bukkit.createBlockData(loc.clone().add(ironBlockLocations[i][0], -3, ironBlockLocations[i][1]).getBlock().getType()));
-		}
-
-		SiegeWar.getSiegeWar().getScheduler().runLater(player, () -> changeBlocksBackToNormal(player,loc), 1l);
+		SiegeWar.getSiegeWar().getScheduler().runLater(player, () -> {
+			player.sendBlockChange(loc.clone().subtract(0, 1, 0), Bukkit.createBlockData(loc.clone().subtract(0, 1, 0).getBlock().getType()));
+			player.sendBlockChange(loc.clone().subtract(0, 2, 0), Bukkit.createBlockData(loc.clone().subtract(0, 2, 0).getBlock().getType()));
+
+			int[][] ironBlockLocations = {{1, 1}, {1, 0}, {1, -1}, {0, 1}, {0, 0}, {0, -1}, {-1, 1}, {-1, 0}, {-1, -1}};
+			for (int i = 0; i < 9; i++) {
+				player.sendBlockChange(loc.clone().add(ironBlockLocations[i][0], -3, ironBlockLocations[i][1]), Bukkit.createBlockData(loc.clone().add(ironBlockLocations[i][0], -3, ironBlockLocations[i][1]).getBlock().getType()));
+			}
+			changeBlocksBackToNormal(player,loc);
```

Comment:
❓ 
- Why was the scheduler call removed here?
- As I understand it, that was added just recently to prevent an exception in Folia.

Reply:
i moved all of it into the scheduler call, before only `changeBlocksBackToNormal(player,loc), 1l);` was being run on the region thread and everything else was on the global thread, this moves all of it to the region thread and prevents the exception